invenio_rdm_records/collections/service.py

```diff
@@ -131,3 +162,43 @@ def read_logo(self, identity, slug):
         if _exists:
             return url_for("static", filename=logo_path)
         raise LogoNotFoundError()
+
+    def read_many(self, identity, ids_=None, depth=2):
+        """Get many collections."""
+        self.require_permission(identity, "read")
+
+        ids_ = ids_ or []
+        res = self.collection_cls.resolve_many(ids_, depth=depth)
+        return CollectionList(
+            identity, res, self.collection_schema, None, self.links_item_tpl
+        )
+
+    def read_all(self, identity, depth=2):
+        """Get all collections."""
+        self.require_permission(identity, "read")
+        res = self.collection_cls.resolve_all(depth=depth)
+        return CollectionList(
+            identity, res, self.collection_schema, None, self.links_item_tpl
+        )
+
+    def search_records(self, identity, collection_or_id, params=None):
```

Comment:
nit: just for clarity of usage
```suggestion
    def search_collection_records(self, identity, collection_or_id, params=None):
```